implementations/rust/ockam/ockam_abac/src/policy/policies.rs

```diff
@@ -0,0 +1,219 @@
+use crate::attribute_access_control::{ABAC_HAS_CREDENTIAL_KEY, SUBJECT_KEY};
+use crate::policy::ResourceTypePolicy;
+use crate::{
+    Action, Env, Expr, PolicyAccessControl, Resource, ResourceName, ResourcePoliciesRepository,
+    ResourcePolicy, ResourceToResourceTypeRepository, ResourceType, ResourceTypePoliciesRepository,
+};
+use ockam_core::compat::format;
+use ockam_core::compat::sync::Arc;
+use ockam_core::compat::vec::Vec;
+use ockam_core::Result;
+use ockam_identity::{Identifier, IdentitiesAttributes};
+use tracing::{debug, instrument};
+
+#[derive(Clone)]
+pub struct Policies {
+    resources_repository: Arc<dyn ResourcePoliciesRepository>,
+    resource_types_repository: Arc<dyn ResourceTypePoliciesRepository>,
+    resource_to_resource_type_repository: Arc<dyn ResourceToResourceTypeRepository>,
+}
+
+impl Policies {
+    pub fn new(
+        resources_repository: Arc<dyn ResourcePoliciesRepository>,
+        resource_types_repository: Arc<dyn ResourceTypePoliciesRepository>,
+        resource_to_resource_type_repository: Arc<dyn ResourceToResourceTypeRepository>,
+    ) -> Self {
+        Self {
+            resources_repository,
+            resource_types_repository,
+            resource_to_resource_type_repository,
+        }
+    }
+
+    #[instrument(skip_all, fields(resource = %resource, action = %action, env = %env, authority = %authority))]
+    pub async fn make_policy_access_control(
+        &self,
+        identities_attributes: Arc<IdentitiesAttributes>,
+        resource: Resource,
+        action: Action,
+        env: Env,
+        authority: Identifier,
+    ) -> Result<PolicyAccessControl> {
+        debug!(
+            "set a policy access control for resource '{}' of type {} and action '{}'",
+            &resource.resource_name, &resource.resource_type, &action
+        );
+        Ok(PolicyAccessControl::new(
+            self.clone(),
+            identities_attributes,
+            authority,
+            env,
+            resource,
+            action.clone(),
+        ))
+    }
+}
+
+// Methods for resource policies
+impl Policies {
+    pub async fn set_policy_for_resource(
+        &self,
+        resource: &Resource,
+        action: &Action,
+        expression: &Option<Expr>,
+    ) -> Result<()> {
+        // Set policy for resource name if an expression is provided.
+        if let Some(expression) = expression {
+            self.resources_repository
+                .set_policy(&resource.resource_name, action, expression)
+                .await?;
+        }
+        // Set resource to resource type mapping.
+        self.resource_to_resource_type_repository
+            .set(resource)
+            .await?;
+        // Set default policy for resource_type if none exists.
+        if self
```

Comment:
Can this be a sqlx migration that on the startup sets default expression for every resource type?

Reply:
If we prepopulate this table from a migration we will be coupling the SQL code with the values of both `ResourceType` and `Action`. We don't even have a defined set of values for `Action`. So, every time we add a new value for any of those objects, we'd need a new migration to add them to the table. I think that's prone to errors and harder to maintain.

Reply:
Your points are valid, but I still would consider that option. Doing it right here is 1. Inefficient 2. Can lead to it not being called at all (what if we didn't call `store_policy_for_resource` for that resource? the default policy won't be set for that resource type at all?)
That being said, the migration is not perfect place, but AFAIK the goal is to set default policies for all resource types in the very beginning of the program execution, right? That's what sql migration could do

Reply:
The alternative could be to return a default policy for a resource type in source code inside the `get_policy_for_resource_type`, but I like it less

Reply:
Maybe we can do both?
- Adding a known set of resource types in the migration
- Keep the logic in `store_policy_for_resource`

Reply:
Another problem I didn't mention of adding values to the migration. The table has a `node_name` column.

Reply:
We'll attempt insert the default policies on node creation

Reply:
@SanjoDeundiak ready!